### ISSUE TITLE
identify method interface vs class type mismatch

### ISSUE THREAD

**Issue by 509dave16** [segmentio/analytics-react-native#531]
The Segment [Spec: Identify docs](https://segment.com/docs/connections/spec/identify/#identities) mention that if using an `anonymousId` you can call identify  without specifying a userId (i.e. it can be `undefined`).

But in the NPM package, there's a conflict between `ClientMethods` interface and the `SegmentClient` class:
`ClientMethods` interface does not allow for `userId` argument to be `undefined` (i.e. it's required): https://github.com/segmentio/analytics-react-native/blob/d48b7f9722399a94acbae2124972abc0169d9cac/packages/core/src/types.ts#L131
`SegmentClient` class does allow for `userId` argument to be undefined (i.e. it's optional): https://github.com/segmentio/analytics-react-native/blob/d48b7f9722399a94acbae2124972abc0169d9cac/packages/core/src/analytics.ts#L495

**TLDR;** Cannot make `identify` calls for anonymous users when leveraging the `useAnalytics` hook. This is a blocker for customers that want to identify traits for anonymous users while at the same time leveraging the `useAnalytics` hook.

- `analytics-react-native` version: *2.1.12*
- Integrations versions (if used): N/A
- React Native version: 0.66.4
- iOS or Android or both? Both

**Steps to reproduce**
Attempt to use `identify` method on client returned from `createClient` module export when surfaced via the `useAnalytics` React hook.

**Expected behavior**
`userId` argument should be optional for `identify` method of `ClientMethods` interface such that `undefined` could be passed to allow for updating traits of anonymous users when leveraging the `useAnalytics` hook.

**Actual behavior**
`userId` argument is required for `identify` method of `ClientMethods` return value of `useAnalytics` hook.

**Comment by oscb** [comment]
@509dave16 Thanks for the report, this is a good find. I'm working on the next release this week with some fixes and some new features, I'll include a fix for these types.

**Comment by 509dave16** [comment]
Awesome! Thanks @oscb !!

**Comment by 509dave16** [comment]
@oscb Sounds like a plan! Really appreciate your work on this, Oscar!!
